src/python/grpcio_tests/tests_aio/unit/server_test.py

```diff
@@ -71,148 +88,144 @@ def service(self, handler_details):
 
 class TestServer(AioTestBase):
 
-    def test_unary_unary(self):
-
-        async def test_unary_unary_body():
-            result = await _start_test_server()
-            server_target = result[0]
-
-            async with aio.insecure_channel(server_target) as channel:
-                unary_call = channel.unary_unary(_SIMPLE_UNARY_UNARY)
-                response = await unary_call(_REQUEST)
-                self.assertEqual(response, _RESPONSE)
-
-        self.loop.run_until_complete(test_unary_unary_body())
-
-    def test_shutdown(self):
-
-        async def test_shutdown_body():
-            _, server, _ = await _start_test_server()
-            await server.stop(None)
-
-        self.loop.run_until_complete(test_shutdown_body())
+    async def setUp(self):
+        self._server_target, self._server, self._generic_handler = await _start_test_server(
+        )
+
+    async def tearDown(self):
+        await self._server.stop(None)
+
+    async def test_unary_unary(self):
+        async with aio.insecure_channel(self._server_target) as channel:
+            unary_unary_call = channel.unary_unary(_SIMPLE_UNARY_UNARY)
+            response = await unary_unary_call(_REQUEST)
+            self.assertEqual(response, _RESPONSE)
+
+    async def test_unary_stream_async_generator(self):
+        async with aio.insecure_channel(self._server_target) as channel:
+            unary_stream_call = channel.unary_stream(_UNARY_STREAM_ASYNC_GEN)
+            call = unary_stream_call(_REQUEST)
+            await self._generic_handler.wait_for_call()
```

Comment:
I'm not sure I understand why you have to await here and then also with the `asyncio.wait_for` that immediately follows.

Reply:
This is a due to a smart (horrible) fast-forward git merge... Removed.